@xen-orchestra/backups/_runners/_writers/IncrementalXapiWriter.mjs

```diff
@@ -103,10 +104,12 @@ export class IncrementalXapiWriter extends MixinXapiWriter(AbstractIncrementalWr
       .filter(_ => !!_)
     // @todo use index ?
 
-    const replicatedVdis = sr.$VDIs.filter(({ other_config }) => {
-      // REPLICATED_TO_SR_UUID is not used here since we are already filtering from sr.$VDIs
-      return sourceVdiUuids.includes(other_config?.[COPY_OF])
-    })
+    const replicatedVdis = sr.$VDIs
+      .filter(_ => !!_) // some VDIs are undefined
+      .filter(({ other_config }) => {
+        // REPLICATED_TO_SR_UUID is not used here since we are already filtering from sr.$VDIs
+        return sourceVdiUuids.includes(other_config?.[COPY_OF])
+      })
```

Comment:
Instead of doing a double iteration, why don't you filter everything in the same iteration?

```js
sr.$VDIs
   .filter(vdi => vdi !== undefined && sourceVdiUuids.includes(vdi.other_config?.[COPY_OF]))
```